crates/ruff_python_formatter/tests/snapshots/format@statement__match.py.snap

```diff
@@ -124,6 +155,29 @@ def foo():
     match inside_func:  # comment
         case NOT_YET_IMPLEMENTED_Pattern:
             pass
+
+
+match newlines:
+    # case 1 leading comment
+
+    case NOT_YET_IMPLEMENTED_Pattern:  # case dangling comment
+        # pass leading comment
+        pass
+        # pass trailing comment
+
+    # case 2 leading comment
+
+    case NOT_YET_IMPLEMENTED_Pattern if foo == 2:  # second
+        pass
+
+    case NOT_YET_IMPLEMENTED_Pattern if (foo := 1):  # third
```

Comment:
These parentheses are preserved in black as well.

Reply:
Oh actually, even if the line length is more than 88, this adds the parentheses thus not breaking on multiple lines. I'll have to look into it as using `Parenthesize::IfRequired` isn't correct because otherwise the parentheses in this example won't be preserved.

Reply:
This is solved. The heuristic is that if the expression already had parentheses, preserve that otherwise don't do anything even if it goes beyond the line length.

Reply:
You may want to use `Parenthesize::Optional` which preserves Parentheses. Whether you want to use `guard.format()` or `maybe_parenthesize` depends on if parentheses should always be added if they don't fit on the line or only if they don't bring their own parentheses (`e.g. [a, b, c, d]` doesn't need addtional parentheses because it has its own). 

I would expect it to be `maybe_parenthesize` because it is used in all positions where expressions are formatted inside a statement.

Reply:
`Parenthesize::Optional` would break the expression into multiple lines which black doesn't do:

```python
match long_lines:
    case "this is a long line for if condition" if aaaaaaaaahhhhhhhh == 1 and bbbbbbaaaaaaaaaaa == 2:  # comment
        pass
```

Running black on the above snippet wouldn't give any difference but if we use `Optional`, then we parenthesize the `if` expression:

```python
match long_lines:
    case NOT_YET_IMPLEMENTED_Pattern if (
        aaaaaaaaahhhhhhhh == 1 and bbbbbbaaaaaaaaaaa == 2
    ):  # comment
        pass
```